main.py

```diff
@@ -2,23 +2,90 @@
 #
 # Simple manager prototype for xqemu
 #
-from PyQt5.QtWidgets import QApplication, QDialog, QFileDialog, QMainWindow, QMessageBox
+from PyQt5.QtWidgets import QApplication, QDialog, QFileDialog, QMainWindow, QMessageBox, QTableWidgetItem, QAbstractItemView, QHeaderView
 from PyQt5.uic import loadUiType
 from PyQt5 import QtCore, QtGui
 from qmp import QEMUMonitorProtocol
+from glob import glob1
 import sys
 import os, os.path
 import json
 import subprocess
 import time
 import platform
+import csv
+import sys
 
 SETTINGS_FILE = './settings.json'
+LIBRARY_CACHE = './library.cache'
 
 # Load UI files
 settings_class, _ = loadUiType('settings.ui')
 mainwindow_class, _ = loadUiType('mainwindow.ui')
 
+class DiscImageFormat(object):
+	XISO = 'XISO'
+	ISO9660 = 'ISO9660'
+	Unknown = 'Unknown format'
+
+def read_from_offset(file, offset, length):
+	file.seek(offset)
+	return file.read(length).decode('unicode-escape')
```

Comment:
Do we want this?

I assume we want:

```
def read_from_offset(file, offset, length, encoding='ascii'):
	file.seek(offset)
	return file.read(length).decode(encoding)
```

Maybe not have an encoding default at all, to avoid lazyness + increate code readability for caller. Optionally, return bytes object and handle decoding in caller.

Reply:
`ascii` will error since on some files (some i used for testing) it wont be able to decode the bytes due to out of range.
Better would be `unicode-escape` but ya we should make it a parameter with default value